src/lib/featureFlags/featureFlags.ts

```diff
@@ -0,0 +1,43 @@
+import {
+  FeatureFlagConfig,
+  FeatureFlagProvider,
+} from "./featureFlagProviderShared"
+
+const registeredFeatureFlagProviders = new Map<symbol, any>()
+
+export async function createFeatureFlagProvider<T extends FeatureFlagConfig>(
```

Comment:
Another naming thing is: in React, there's a standard context "Provider" naming pattern, and doubling up between there and here is slightly confusing. 

Is there another term that we can use use so that we're not sharing conflicting conceptual vocabulary? `featureFlagManager`, `featureFlagController`, or something like that?

Reply:
Good point. Changed to `featureFlagService`